### PR TITLE
New behaviour/screen for My Lists

### PR DESCRIPTION
**Pull request by cooltey**
https://phabricator.wikimedia.org/T256775

### CODE REVIEW

File: app/src/main/java/org/wikipedia/views/MessageCardView.kt [hunk 1]
```diff
@@ -4,17 +4,43 @@ import android.content.Context
...
+
+    fun setMessageTitle(title: String) {
+        messageTitleView.text = title
+    }
+
+    fun setMessageText(text: String) {
+        messageTextView.text = text
+    }
+
+    fun setImageSource(imageSource: Int) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/views/MessageCardView.kt:29]: This should probably be named `setImageResource`, and annotate the parameter with `@DrawableRes`.

File: app/src/main/java/org/wikipedia/views/MessageCardView.kt [hunk 2]
```diff
@@ -4,17 +4,43 @@ import android.content.Context
...
+        if (imageSource < 0) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/views/MessageCardView.kt:30]: It's not very obvious to implicitly show/hide the image based on the resource parameter.  Perhaps we can have a second (boolean) parameter to show/hide the image.

File: app/src/main/java/org/wikipedia/views/MessageCardView.kt [hunk 3]
```diff
@@ -4,17 +4,43 @@ import android.content.Context
...
+            imageView.visibility = View.GONE
+        } else {
+            imageView.visibility = View.VISIBLE
+            imageView.setImageResource(imageSource)
+        }
+    }
+
+    fun setButton(text: String, listener: OnClickListener, applyListenerToContainer: Boolean) {
```
Comment (dbrant) [inline app/src/main/java/org/wikipedia/views/MessageCardView.kt:38]: Can the `text` be an Int, annotated with `@StringRes`?

File: app/src/main/java/org/wikipedia/views/MessageCardView.kt (hunk 1 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/views/MessageCardView.kt]: Makes sense. Done!

File: app/src/main/java/org/wikipedia/views/MessageCardView.kt (hunk 2 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/views/MessageCardView.kt]: You're correct. Done!

File: app/src/main/java/org/wikipedia/views/MessageCardView.kt (hunk 3 above)
Comment (cooltey) [reply app/src/main/java/org/wikipedia/views/MessageCardView.kt]: Sure! Done.

**Comment by cooltey** [comment]
Pending discussion with Robin: https://phabricator.wikimedia.org/T256775#6283622

**Pull request merged**